File: csharp/log-levels/LogLevels.cs
using System;

static class LogLine
{
    public static string Message(string logLine)
    {
        var idx = logLine.IndexOf(':');
        return logLine.Substring(idx+1).Trim();
    }

    public static string LogLevel(string logLine)
    {
        var idx = logLine.IndexOf(':');
        var level = logLine.Substring(0, idx);
        level = level.Replace("[", "");
        level = level.Replace("]", "");
        return level.ToLower();
    }

    public static string Reformat(string logLine)
    {
        var msg = Message(logLine);
        var lvl = LogLevel(logLine);
        return $"{msg} ({lvl})";
    }
}

File: cpp/log-levels/log_levels.cpp
#include <string>

using namespace std;

namespace log_line
{
    string message(string log_line)
    {
        auto delimiter = log_line.find(':');
        return log_line.substr(delimiter + 2);
    }
    string log_level(string log_line)
    {
        auto delimiter = log_line.find(']');
        return log_line.substr(1, delimiter - 1);
    }
    string reformat(string log_line)
    {
        return message(log_line) + " (" + log_level(log_line) + ")";
    }
} // namespace log_line
